shumai/util/memory.ts

```diff
@@ -0,0 +1,73 @@
+import { Tensor } from '../tensor/tensor'
+export const creationTracker: Map<number, Tensor> = new Map()
+export const altTracker: Record<number, Tensor> = {}
+
+export let is_tidy_run = false
```

Comment:
as mentioned above, we can unify this with `creationTracker` maybe as `_tidyTracker`

Reply:
Ahh, I like this; keeps it a bit cleaner! (and I prefer the name `_tidyTracker` better as well for sure)